src/Files.App/Shell/Win32API.cs

```diff
@@ -362,7 +469,34 @@ public static bool SetCustomDirectoryIcon(string folderPath, string iconFile, in
             fcs.cchIconFile = 0;
             fcs.iIconIndex = iconIndex;
 
-            return Shell32.SHGetSetFolderCustomSettings(ref fcs, folderPath, Shell32.FCS.FCS_FORCEWRITE).Succeeded;
+            var success = Shell32.SHGetSetFolderCustomSettings(ref fcs, folderPath, Shell32.FCS.FCS_FORCEWRITE).Succeeded;
+            if (success)
+            {
+                _iconAndOverlayCache[folderPath] = new();
+            }
+            return success;
+        }
+
+        public static async Task<bool> SetCustomFileIconAsync(string filePath, string iconFile, int iconIndex = 0)
+        {
+            var connection = await AppServiceConnectionHelper.Instance;
+            if (connection != null)
+            {
```

Comment:
```suggestion
            if (connection == null)
                        return;
```
Returning early can help make the code more readable 👍